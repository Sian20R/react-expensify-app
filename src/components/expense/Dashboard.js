import React from 'react';
import ExpenseList from './../expense/List';
import ExpenseListFilters from './../expense/ListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);
export default ExpenseDashboardPage;