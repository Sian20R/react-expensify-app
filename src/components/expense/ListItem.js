import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral'; 

export const ExpenseItem = ({id, description, amount, createdAt}) => (
      <Link className="list-item" to={`/edit/${id}`}>
        <div>
          <h3 className="list-item__title">{description}</h3>
          <span className="list-item__sub-title">{moment(createdAt).format('Do MMMM YYYY')}</span>
        </div>
        <h3 className="list-item__content">{numeral(amount).format('$0,0.00')}</h3>
      </Link>    
);

const mapStateToProps = (state) => {
  return {
      expenses: state.expenses
  };
};

export default connect(mapStateToProps)(ExpenseItem);