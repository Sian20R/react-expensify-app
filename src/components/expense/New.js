import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../../actions/expenses';
import ExpenseForm from '../expense/Form';

export class NewExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.newExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>New Expense</h1>
                <ExpenseForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    } 
};

const mapDispatchToProps = (dispatch) => ({
    newExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(NewExpensePage);