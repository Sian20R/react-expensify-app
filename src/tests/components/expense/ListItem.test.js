import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseItem } from '../../../components/expense/ListItem'; 
import expenses from '../../fixtures/expenses';

test('should render ExpenseListItem with fixture data', () => {
    const wrapper = shallow(<ExpenseItem {...expenses[1]} />);
    expect(wrapper).toMatchSnapshot();
});
 