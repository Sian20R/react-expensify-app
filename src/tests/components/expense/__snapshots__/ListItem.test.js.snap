// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with fixture data 1`] = `
<Link
  className="list-item"
  to="/edit/2"
>
  <div>
    <h3
      className="list-item__title"
    >
      Rent
    </h3>
    <span
      className="list-item__sub-title"
    >
      28th December 1969
    </span>
  </div>
  <h3
    className="list-item__content"
  >
    $109,500.00
  </h3>
</Link>
`;
